array-manipulation-1/src/reverse.js

```diff
@@ -0,0 +1,16 @@
+/* exported reverse */
+/*
+need a result array
+use a for loop
+  start index at 0
+  append array at array.length - 1 - i to result array
+  stop when index is larger than array length
+return result
+*/
+function reverse(array) {
+  var result = [];
+  for (var i = 0; i < array.length; i++) {
+    result.push(array[array.length - 1 - i]);
```

Comment:
You can simplify this considerably by just making the loop go over the array in reverse